pkg/fanal/analyzer/os/redhatbase/redhatbase.go

```diff
@@ -72,6 +75,46 @@ func (a redhatOSAnalyzer) parseRelease(r io.Reader) (types.OS, error) {
 				Family: types.Fedora,
 				Name:   result[2],
 			}, nil
+		case "red hat enterprise linux coreos":
+			// https://access.redhat.com/articles/6907891
+
+			var major, minor, rel int
+
+			split := strings.SplitN(result[2], ".", 4)
+			major, _ = strconv.Atoi(split[0])
+			if len(split) > 1 {
+				minor, _ = strconv.Atoi(split[1])
+			}
+			if len(split) > 2 {
+				rel, _ = strconv.Atoi(split[2])
+			}
```

Comment:
Can you add a comment with examples why we need this split?
It seems we can just use semver.Parse on `result[2]`